lib/utils/log/formatter_test.go

```diff
@@ -379,8 +379,39 @@ func BenchmarkFormatter(b *testing.B) {
 
 			for i := 0; i < b.N; i++ {
 				l := logger.With("test", 123).With("animal", "llama\n").With("error", logErr)
-				l.With(trace.ComponentFields, fields).Info(message, "diag_addr", &addr)
+				l.With(trace.ComponentFields, fields).InfoContext(ctx, message, "diag_addr", &addr)
 			}
 		})
 	})
 }
+
+func TestConcurrentOutput(t *testing.T) {
+	t.Run("logrus", func(t *testing.T) {
+		debugFormatter := NewDefaultTextFormatter(true)
+		require.NoError(t, debugFormatter.CheckAndSetDefaults())
+		logrus.SetFormatter(debugFormatter)
+		logrus.SetOutput(os.Stdout)
+
+		logger := logrus.WithField(trace.Component, "test")
+
+		for i := 0; i < 1000; i++ {
+			go func(i int) {
```

Comment:
Use a WaitGroup and wait for all goroutines to return before completing the test.